### ISSUE TITLE
problems with `IntermediateGroup` and `IntermediateSubgroups`

### ISSUE THREAD

**Issue by ThomasBreuer** [gap-system/gap#4246]
A recent message to the GAP Forum asked about an explanation of a `fail` result of `IntermediateGroup`.
```
gap> g:= PSL( 2, 7^3 );;                                            
gap> s:= SylowSubgroup( g, 2 );;
gap> Index( g, s );
2522079
gap> IntermediateGroup( g, s );
fail
```
(Note that the Sylow 2-subgroups in PSL(2, 7^3) are *not* maximal.)

If one reads the documentation of `IntermediateGroup` carefully then this is not a bug.
However, I would vote for making the documentation more explicit, in the sense that `fail` is returned whenever the function does not find an intermediate group, and that the maximality of the given subgroup is only one possible reason.

In the case of `IntermediateSubgroups`, the documentation is explicit.
Thus the following is a bug.
```
gap> IntermediateSubgroups( g, s );
rec( inclusions := [  ], subgroups := [  ] )
```
I think that the problem in this case is that the `IntermediateSubgroups` method in question calls `TryMaximalSubgroupClassReps` (which is undocumented) and checks for a `fail` result, but the result is an empty list.

**Comment by hulpke** [comment]
This has been fixed as part of #4219 I thus close.
(The new code now does not return `fail' but will try hard, giving a warning that it could be costly. Since `IntermediateGroup` is not used in the library, this change should have no effect.

**Comment by ThomasBreuer** [comment]
Thanks, @hulpke, for the code fixes. I can confirm that the reported problems have disappeared after the changes, in the examples which I had found.
